src/main/java/org/jitsi/videobridge/octo/ConfOctoTransport.java

```diff
@@ -0,0 +1,561 @@
+    }
+
+    public ConfOctoTransport(Conference conference, Clock clock)
+    {
+        this.conference = conference;
+        this.clock = clock;
+        this.logger = conference.getLogger().createChildLogger(this.getClass().getName());
+        BundleContext bundleContext = conference.getBundleContext();
+        OctoRelayService octoRelayService
+            = bundleContext == null ? null :
+            ServiceUtils2.getService(bundleContext, OctoRelayService.class);
+
+        if (octoRelayService == null)
+        {
+            throw new IllegalStateException("Couldn't get OctoRelayService");
+        }
+
+        bridgeOctoTransport = octoRelayService.getBridgeOctoTransport();
+        if (bridgeOctoTransport == null)
+        {
+            throw new IllegalStateException("Couldn't get OctoTransport");
+        }
+
+        octoEndpoints = new OctoEndpoints(conference);
+        octoTransceiver = new OctoTransceiver("tentacle-" + conference.getGid(), logger);
+        octoTransceiver.setIncomingPacketHandler(conference::handleIncomingPacket);
+        octoTransceiver.setOutgoingPacketHandler(packetInfo ->
+        {
+            throw new RuntimeException("This should not be used for sending");
+        });
+
+        // Some remote packets didn't originate from an endpoint, but from
+        // the bridge (like keyframe requests).
+        addHandler(JVB_EP_ID, new IncomingOctoEpPacketHandler()
+        {
+            @Override
+            public void handleIncomingPacket(@NotNull OctoPacketInfo packetInfo)
+            {
+                octoTransceiver.handleIncomingPacket(packetInfo);
+            }
+        });
+    }
+
+    /**
+     * Adds a {@link PayloadType}
+     */
+    public void addPayloadType(PayloadType payloadType)
+    {
+        if (running.get())
+        {
+            octoEndpoints.addPayloadType(payloadType);
+        }
+    }
+
+    /**
+     * Sets the list of remote relays to send packets to.
+     * @param relays the list of relay IDs, which are converted to addresses
+     * using the logic in {@link OctoUtils}.
+     */
+    public void setRelays(Collection<String> relays)
+    {
+        if (!running.get())
+        {
+            return;
+        }
+        Objects.requireNonNull(
+            bridgeOctoTransport,
+            "Octo requested but not configured");
+
+        Set<SocketAddress> socketAddresses = new HashSet<>();
+        for (String relay : relays)
+        {
+            SocketAddress socketAddress = OctoUtils.Companion.relayIdToSocketAddress(relay);
+            if (socketAddress != null)
+            {
+                socketAddresses.add(socketAddress);
+            }
+        }
+
+        setTargets(socketAddresses);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean wants(PacketInfo packetInfo)
+    {
+        // Cthulhu devours everything (as long as it's not coming from
+        // itself, and we have targets).
+        return running.get() &&
```

Comment:
This is probably premature optimization, but...

Most rejected packets will come from the `instanceof` check. Do we want to put it in front to avoid some calls to `AtomicBoolean.get()` (coming from multiple threads, iirc)?

Reply:
I can do that...if it makes you feel any better `AtomicBoolean` calls didn't come up at all in a 5 minute, 20 person conference (split across 2 bridges, sampled from one).  But I did CPU sampling, not tracing.  I can run a trace real quick and get more data.

Reply:
I tracked this and it looks like it gets inlined entirely, so I think we're good.